pkg/agent/manager/cache/cache.go

```diff
@@ -84,10 +121,24 @@ func (c *cacheImpl) Entry(regEntry *common.RegistrationEntry) *Entry {
 func (c *cacheImpl) SetEntry(entry *Entry) {
 	c.m.Lock()
 	defer c.m.Unlock()
+	subs := c.Subscribers.Get(entry.RegistrationEntry.Selectors)
+	for _, sub := range subs {
+		var entries []Entry
+		for _, e := range c.cache {
```

Comment:
Might be good to move this logic into an unexported function like `buildWorkloadUpdate(*subscriber)`